### ISSUE TITLE
CloudWatch with Ids / MetricsMath is broken by 5.3.2 release

### ISSUE THREAD

**Issue by erothmayer** [grafana/grafana#13939]
Please include this information:
### What Grafana version are you using?
5.3.2

### What datasource are you using?
CloudWatch Metrics

### What OS are you running grafana on?
Stock Docker distribution

### What did you do?
1. Upgraded to 5.3.2
2. Configure CloudWatch Metrics data source
3. Add 2 metrics and confirm they are graphed correctly
4. Add an id to each of them so they can be referenced by a Metric Math expression
5. Add a metric math expression which references them

### What was the expected result?
All 3 data series return and render (as is the case running on 5.3.1):

![image](https://user-images.githubusercontent.com/26010650/47886726-1133c200-ddf9-11e8-97a0-2d61dca90998.png)

### What happened instead?
As the id is added to each metric, it disappears from the graph. Nothing displays.

![image](https://user-images.githubusercontent.com/26010650/47886739-27418280-ddf9-11e8-8047-693836fd6425.png)

As noted above, I confirmed this was working in 5.3.1, but not in 5.3.2. Looking through the changelog, it seems probable that this was broken by this change: https://github.com/grafana/grafana/issues/13674

Seems like maybe once an id is added to the query, it's conflicting with the refId and hiding the query result from the code above the portions changed in the race condition fix.

Mentioning @mtanda as he seems to have lots of context on this space.

**Comment by torkelo** [comment]
@mtanda any ideas what this could be?

**Comment by mtanda** [comment]
my simple query seems to work in 5.3.2.
I'll investigate this issue in this weekend.

**Comment by erothmayer** [comment]
Apologies, seems this was a bit of a mis-fire on my part.

I tested further this morning and realized I hadn't controlled for a difference in credentials between my two test locations. So, there's still a minor problem, but not the one I thought I had found.

Namely, the MetricMath feature requires an additional permission `cloudwatch:GetMetricData` on the assumed role. When it's not provided, adding an id to any metric causes it to stop displaying.

Could be helpful to note that in the docs, as the 'minimal permissions' don't include this, and it seems sensible to just add to the default.
